Assets/Scripts/Algorithms/Patrolling/Components/RedistributionComponent.cs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2025 MAEPS
+// 
+// This file is part of MAEPS
+// 
+// MAEPS is free software: you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+// 
+// MAEPS is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License along
+// with MAEPS. If not, see http://www.gnu.org/licenses/.
+//
+// Contributors 2025: 
+// Casper Nyvang Sørensen,
+// Christian Ziegler Sejersen,
+// Jakob Meyer Olsen
+// Mads Beyer Mogensen
+using System.Collections.Generic;
+
+using Maes.Map;
+using Maes.Robot;
+
+using UnityEngine;
+
+
+namespace Maes.Algorithms.Patrolling.Components
+{
+    public sealed class RedistributionComponent : IComponent
```

Comment:
What is the difference between PartitionRedistributionComponent and this one? I really want a comment on the components that explain them.

Reply:
I also think the class name should be more specific

Reply:
Just bad naming, have tried improving it.